forest/src/cli/net_cmd.rs

```diff
@@ -0,0 +1,106 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use forest_libp2p::{Multiaddr, Protocol};
+use rpc_api::data_types::AddrInfo;
+use structopt::StructOpt;
+
+use crate::cli::cli_error_and_die;
+
+use super::{handle_rpc_err, print_stdout};
+use rpc_client::net_ops::*;
+
+#[derive(Debug, StructOpt)]
+pub enum NetCommands {
+    /// Lists libp2p swarm listener addresses
+    #[structopt(about = "List listen addresses")]
+    Listen,
+    /// Lists libp2p swarm peers
+    #[structopt(about = "Print peers")]
+    Peers,
+    /// Connects to a peer
+    #[structopt(about = "Connect to a peer by its peer ID and multiaddresses")]
+    Connect {
+        #[structopt(about = "Multiaddr (with /p2p/ protocol)")]
+        address: String,
+    },
+    /// Disconnects from a peer
+    #[structopt(about = "Disconnect from a peer by its peer ID")]
+    Disconnect {
+        #[structopt(about = "Peer ID to disconnect from")]
+        id: String,
+    },
+}
+
+impl NetCommands {
+    pub async fn run(&self) {
+        match self {
+            Self::Listen => match net_addrs_listen(()).await {
+                Ok(info) => {
+                    let addresses: Vec<String> = info
+                        .addrs
+                        .iter()
+                        .map(|addr| format!("{}/p2p/{}", addr.to_string(), info.id.to_string()))
+                        .collect();
+
+                    print_stdout(addresses.join("\n"));
+                }
+                Err(e) => handle_rpc_err(e),
+            },
+            Self::Peers => match net_peers(()).await {
+                Ok(addrs) => {
+                    let output: Vec<String> = addrs
+                        .iter()
+                        .map(|info| {
+                            let addresses: Vec<String> =
+                                info.addrs.iter().map(|addr| addr.to_string()).collect();
+
+                            format!("{}, [{}]", info.id, addresses.join(", "))
+                        })
+                        .collect();
+
+                    print_stdout(output.join("\n"));
+                }
+                Err(e) => handle_rpc_err(e),
+            },
+            Self::Connect { address } => {
+                let addr: Multiaddr = address
+                    .parse()
+                    .expect("parse provided multiaddr from string");
```

Comment:
This would be a better devx as an error printed to stderr

Reply:
And then an exit with the proper exit code

Reply:
Good idea! Will do.